### PR TITLE
GROM-142: Add session data increments

### PR DESCRIPTION
**Pull request by dan-0**

### CODE REVIEW

File: app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationActivity.java [hunk 1]
```diff
@@ -61,6 +69,8 @@ public RegistrationActivity() {
         LocaleUtils.updateConfig(this);
     }
 
+    private RegistrationViewModel viewModel;
```
Comment (huttneab) [inline app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationActivity.java:72]: thought you would have already had to have this VM in this class?

File: app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationActivity.java [hunk 2]
```diff
@@ -157,8 +172,7 @@ private void showCancelDialog() {
         new AlertDialog.Builder(this)
                 .setMessage(R.string.cancel_dialog_message)
                 .setPositiveButton(R.string.dialog_yes, ((dialog, i) -> {
-                    // set the application to abandoned so it gets cleaned up
-
+                    viewModel.incrementAbandonedCount();
```
Comment (huttneab) [inline app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationActivity.java:175]: 👍

File: app/src/main/java/com/rockthevote/grommet/data/db/dao/SessionDao.kt [hunk 3]
```diff
@@ -20,4 +18,7 @@ interface SessionDao {
     @Transaction
     @Query("SELECT * FROM session")
     fun getSessionWithRegistrations(): List<SessionWithRegistrations>
+
+    @Update(onConflict = OnConflictStrategy.REPLACE)
```
Comment (huttneab) [inline app/src/main/java/com/rockthevote/grommet/data/db/dao/SessionDao.kt:22]: I tried this for my insert method and got a unique constraint violation. Wasn't really sure how given that i deleted everything prior to running it.

Does an update method even require a conflict strategy?

Comment (dan-0) [reply app/src/main/java/com/rockthevote/grommet/data/db/dao/SessionDao.kt:22]: Hmm, that's a good point for `Update`, I can't see a case where `onConflict` is needed is needed. I'm not sure why you would have got the constraint violation, was the Primary Key pre-populated?

File: app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationActivity.java (hunk 1 above)
Comment (dan-0) [reply app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationActivity.java:72]: It was being scoped to the Activity from the Fragment, but the Activity itself didn't need it. I think `BaseRegistrationFragment` is the only place where I was getting/creating the VM before

File: app/src/main/java/com/rockthevote/grommet/data/db/dao/SessionDao.kt (hunk 3 above)
Comment (huttneab) [reply app/src/main/java/com/rockthevote/grommet/data/db/dao/SessionDao.kt]: no, don't think so

**Pull request merged**